moveit_kinematics/CMakeLists.txt

```diff
@@ -32,12 +32,12 @@ set(THIS_PACKAGE_INCLUDE_DIRS
 catkin_package(
   LIBRARIES
   INCLUDE_DIRS
-    ${EIGEN3_INCLUDE_DIRS}
     ${THIS_PACKAGE_INCLUDE_DIRS}
   CATKIN_DEPENDS
     pluginlib
     moveit_core
     moveit_ros_planning
+    DEPENDS Eigen3
```

Comment:
@v4hn thanks, you are so kind:)
Great insite  for explaning eigen so detail.
I'm sorry for reply this late. Your advices really help me learn a lot. 
Most people write CmakeLists.txt using template without understanding
what has been happened inside actually. Yes, add Eigen dependencies is out of curiosity.
To tell the truth I may not quite understand the meaning before you give the explanation.

Anyway thank you. Ping me more often please, great teacher. I'll reply it soon next time.:p
(Mailbox full of github notifications...)